tpcc/worker.go

```diff
@@ -172,17 +184,15 @@ func (w *worker) run(errCh chan<- error, warehouseID int) {
 				break
 			}
 		}
-		if *noWait {
-			warehouseID = rand.Intn(*warehouses)
-		} else {
-			sleepTime := float64(t.keyingTime * 1000)
+		if !*noWait {
+			sleepTime := time.Duration(t.keyingTime) * time.Second
 			if firstRun {
 				// Sleep for a random duration up to the keying time to smooth out any
 				// potential thundering herd effects when the load generator starts.
-				sleepTime = sleepTime * rand.Float64()
+				sleepTime = time.Duration(float64(sleepTime)*rand.Float64()) + t.randThinkTime()/2
```

Comment:
this is weird but harmless. why sleep a randThinkTime during the keying time in nowait?